cheetah/track_methods.py

```diff
@@ -56,27 +50,31 @@ def base_rmatrix(
     :param hx: Curvature (1/radius) of the element in 1/m**2.
     :param tilt: Roation of the element relative to the longitudinal axis in rad.
     :param energy: Beam energy in eV.
-    :param device: Device where the transfer matrix is created. If "auto", the device
-        is selected automatically.
     :return: Transfer matrix for the element.
     """
-    if device == "auto":
-        device = "cuda" if torch.cuda.is_available() else "cpu"
+    device = length.device
+    dtype = length.dtype
 
-    tilt = tilt if tilt is not None else torch.tensor(0.0, device=device)
-    energy = energy if energy is not None else torch.tensor(0.0, device=device)
+    tilt = tilt if tilt is not None else torch.tensor(0.0, device=device, dtype=dtype)
+    energy = (
+        energy if energy is not None else torch.tensor(0.0, device=device, dtype=dtype)
+    )
 
```

Comment:
I don't think we need to. This would add to the compute time (if only slightly) and the method is not intended to be used by the user. The Elements that use it should already have checked that the devices are correct, I think.